nectar/downloaders/threaded.py

```diff
@@ -215,31 +227,11 @@ def _fetch(self, request):
                 raise SkipLocation()
 
             _logger.debug("Attempting to connect to {url}.".format(url=request.url))
-
-            for attempt in xrange(self.tries):
-                try:
-                    if attempt > 0:
-                        msg = _("Re-trying {url} due to remote server connection failure.".format(
-                            url=request.url)
-                        )
-                        _logger.warning(msg)
-
-                    response = self.session.get(request.url, headers=headers,
-                                                timeout=(self.config.connect_timeout,
-                                                         self.config.read_timeout))
-
-                    report.headers = response.headers
-                    self.fire_download_headers(report)
-
-                    break
-                except requests.ConnectionError as e:
-                    if isinstance(e.strerror, httplib.BadStatusLine):
-                        msg = _("Download of {url} failed. Re-trying.".format(url=request.url))
-                        _logger.warning(msg)
-                        continue
-                    raise
-            else:
-                raise RetryError(request.url)
```

Comment:
I guess the RetryError() can be removed since it is not used anywhere https://github.com/jeremycline/nectar/blob/connection-error-handling/nectar/downloaders/threaded.py#L51